doc/installingNative.md

```diff
@@ -1,78 +1,100 @@
+* Install [homebrew](https://brew.sh).
 
-* Clone the latest code:
+* Install [miniconda](https://conda.io/docs/user-guide/install/macos.html#).
+
+* Use `git` to clone the PrairieLearn repo (put it wherever you want), and then `cd` to that directory:
 
 ```sh
 git clone https://github.com/PrairieLearn/PrairieLearn.git
 ```
 
-* Install the Node.js libraries:
+Use `git pull` as usual to keep your clone up to date.
+
+* Use homebrew to install [R](https://www.r-project.org):
 
 ```sh
-cd PrairieLearn
-npm install
+brew install r
+```
+
+* Use homebrew to install [PostgreSQL](https://www.postgresql.org):
+
+```sh
+brew install postgresql
 ```
 
-On OS X, it is possible that this process will fail on `node-gyp rebuild` during the install of `mcrypt`. In this case, use the following command instead:
+* Use conda to create a virtual environment (e.g., with name `PL`):
 
 ```sh
-npm install --python=PATH_TO_PYTHON2
+conda create -n PL python=3.6
+conda activate PL
 ```
 
-For example, this might be:
+* Make sure `python`, `python3`, and `python3.6` will all run the right version:
 
 ```sh
-npm install --python=/usr/bin/python2.7
+python --version      # should return "Python 3.6.x"
+python3 --version     # should return "Python 3.6.x"
+python3.6 --version   # should return "Python 3.6.x"
 ```
 
-* Make sure `python3` and `python3.6` will run the right version, and make executable links if needed:
+* Install the python packages that are used on the production server:
 
 ```sh
-python3 --version     # should return "Python 3.6" or higher
-python3.6 --version   # should return "Python 3.6" or higher
+cd PrairieLearn
+python -m pip install -r environments/centos7-plbase/python-requirements.txt
 ```
 
-* Install the Python libraries:
+* (Optionally, install the R packages that are used on the production server, as in `environments/centos7-plbase/r-requirements.R`.)
+
+* Use `npm` to install the node libraries, specifying the use of python 2.7 to avoid failure on `node-gyp rebuild`:
 
 ```sh
 cd PrairieLearn
-python3 -m pip install -r requirements.txt
+npm ci --python=/usr/bin/python2.7
 ```
 
+    (Replace the path to `python2.7` if necessary.)
+
+    Repeat this process every time you `git pull`. It is important to use `npm ci` (which replaces `node_modules` on each call) and not `npm install`, or you will cause changes to `package.json` and/or `package-lock.json`.
+
 * Create the database (one time only):
 
 ```sh
 initdb -D ~/defaultdb
 ```
 
-* Run the database:
+* Run the database (each time you restart your computer):
 
 ```sh
 pg_ctl -D ~/defaultdb -l ~/logfile start
 ```
 
+    Note that you can put `defaultdb` and `logfile` anywhere you like.
+
 * Make sure the `postgres` database user exists and is a superuser (these might error if the user already exists):
 
 ```sh
 psql -c "CREATE USER postgres;"
 psql -c "ALTER USER postgres WITH SUPERUSER;"
```

Comment:
If postgres is installed then you should be able to run `createuser -s postgres` at the commandline to do both of these.